### PR TITLE
Added Implementation of Binary Search Tree

### LINKED ISSUE #8266: Implementation of Binary Search Tree
### Description

Discuss implementation of binary search tree.

### Domain

DSA

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by juhik24**
<hr>

## Description 📜

Implementation of Binary Search Tree.
Fixes #8266 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [ x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ x] DSA
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [ x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [ x] I have performed a self-review of my own code or work.
- [x ] I have commented my code, particularly in hard-to-understand areas.
- [x ] My changes generates no new warnings.
- [x ] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by juhik24** [comment]
Hey @nimisha-yadav , check this PR. This is an implementation of Binary Search Tree.

**Comment by juhik24** [comment]
Hey @nimisha-yadav , check this PR. This is an implementation of Binary Search Tree.

File: DSA/6.1 Trees/Binary_Search_Tree/Binary_Search_Tree.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+## Algorithm to insert an element in BST
+
+* To insert an element in BST, we have to first find the position at which we can insert and we find the position in O(log n) time.
+
+* We will start from root, if the value to be inserted is lesser or equal to root then go to left. 
+
+* If the value is greater and there is no right child then insert as right child or go to right.
+
+![alt text](https://camo.githubusercontent.com/2d18fae81b9c7fb20937de6a1450f257b4589dddaac9e25b38469792d99b8c75/687474703a2f2f6274656368736d617274636c6173732e636f6d2f646174615f737472756374757265732f64735f696d616765732f425354253230436f6e737472756374696f6e2e706e67)
```
Comment (nimisha-yadav) [inline DSA/6.1 Trees/Binary_Search_Tree/Binary_Search_Tree.md:29]: Give credits to the author of the image.

File: DSA/6.1 Trees/Binary_Search_Tree/Binary_Search_Tree.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+       Enter the number to be searched: 22
+       Not Found
+
+**Time Complexity:**
+
+* Time complexity of all BST operations = O(h)
+
+* Here, h = Height of Binary Search Tree
+
```
Comment (nimisha-yadav) [inline DSA/6.1 Trees/Binary_Search_Tree/Binary_Search_Tree.md:124]: Add references

**Review by nimisha-yadav — COMMENTED** [review]
Also, rename your folder as 3. Binary Search Tree and rename the file as README.md

**Comment by juhik24** [comment]
> Also, rename your folder as 3. Binary Search Tree and rename the file as README.md

Done

**Review by nimisha-yadav — APPROVED** [review]

**Pull request merged**